### ISSUE TITLE
Editor crashes while sending data

### ISSUE THREAD

**Issue by ziliwap** [getnamo/socketio-client-ue4#65]
The editor crashes while sending data from UE:
![2017-11-06 16_06_21-firstpersoncharacter](https://user-images.githubusercontent.com/24809138/32469501-41bf5fb0-c310-11e7-9a95-524a2d3aa478.png)
I tried both, the "emit" function and the "emit with callback" function. I have a function in place with correct input signatures to receive the callback in case of "emit with callback".

I am receiving the data correctly on the server but I believe its the acknowledgement message handling, that is making the editor crash. I am acknowledging with :
"EventCallback",{"data":"sample text"}

**Comment by getnamo** [comment]
To send ```{"data":"sample text"}``` js object as a callback, your server side javascript should look something like this

```js
io.on('connection', function(socket){
	//...

	//respond to 'my event' on the server
	socket.on('my event', function(message, callback){

		//make an object with {"data":"sample text"}
		let reponseObject = {};
		responseObject['data'] = 'sample text';

		//make a callback function call with the data
		callback(responseObject);	
	});
}
```

Then ensure you have a valid callback function that matches your function name with SIOJsonValue as it's first parameter (with the function name ```EventCallback``` instead of ```OnEcho``` as per your setup)
![img](https://camo.githubusercontent.com/0e205b51dc669c6a0d9ffe0c60f77c7edeb1d59d/687474703a2f2f692e696d6775722e636f6d2f5051696e4459792e676966)
see readme section for details: https://github.com/getnamo/socketio-client-ue4#emit-with-callback

If both of those things are correct and you're still getting an error, please provide plugin & ue version and any crash logs under saved/logs

**Comment by ziliwap** [comment]
I am still experiencing the crash with the following log:

Fatal error: [File:D:\Build\++UE4+Release-4.17+Compile\Sync\Engine\Source\Runtime\Core\Private\GenericPlatform\GenericPlatformMemory.cpp] [Line: 174] 
Ran out of memory allocating 18446744072482788960 bytes with alignment 0


KERNELBASE.dll!0x00000000B10F9E08
UE4Editor-Core.dll!0x00000000687914B4
UE4Editor-Core.dll!0x000000006857AD5B
UE4Editor-Core.dll!0x0000000068510179
UE4Editor-Core.dll!0x00000000683DAFD5
UE4Editor-Core.dll!0x0000000068454062
UE4Editor-Core.dll!0x000000006845426E
UE4Editor-Core.dll!0x000000006837ABE6
UE4Editor-Core.dll!0x0000000068359ACC
UE4Editor-Core.dll!0x000000006835D51E
UE4Editor-CoreUObject.dll!0x00000000740C52BD
UE4Editor-CoreUObject.dll!0x00000000743AD488
UE4Editor-Engine.dll!0x000000006A2DFD5F
UE4Editor-CoreUObject.dll!0x00000000741AEF44
UE4Editor-CoreUObject.dll!0x000000007436DF94
UE4Editor-CoreUObject.dll!0x00000000743832F1
UE4Editor-CoreUObject.dll!0x00000000743ACAF0
UE4Editor-CoreUObject.dll!0x00000000743854D2
UE4Editor-CoreUObject.dll!0x00000000741AEF44
UE4Editor-CoreUObject.dll!0x0000000074384865
UE4Editor-Engine.dll!0x0000000069A3B217
UE4Editor-SocketIOClient.dll!0x0000000054E1EEBB
UE4Editor-SocketIOClient.dll!0x0000000054E1BB0D
UE4Editor-SocketIOClient.dll!0x0000000054E1A909
UE4Editor-SocketIOClient.dll!0x0000000054E231C5
UE4Editor-Core.dll!0x00000000683E24F5
UE4Editor-Core.dll!0x00000000683E27F0
UE4Editor-Engine.dll!0x000000006AAFBA8A
UE4Editor.exe!0x00000000EF696BDA
UE4Editor.exe!0x00000000EF6A3970
UE4Editor.exe!0x00000000EF6A39EA
UE4Editor.exe!0x00000000EF6B02D9
UE4Editor.exe!0x00000000EF6B1C47
KERNEL32.DLL!0x00000000B2A92774
ntdll.dll!0x00000000B4A50D51
ntdll.dll!0x00000000B4A50D51

I am using Flask (have to due to requirements) for the server and the code that I am using is: 

```
@socketio.on('my event')
def echo(message):
    print('My event : ' + str(message))
    return 'EventCallbackString'

```
I tried cross checking on a browser and I am getting the acknowledgement correctly using the following code:

```
function clicked() {
    console.log('clicked');

    var resp = socket.emit('my event', {'data':"Some text"}, function name(params) {
    console.log(params);
    });
    console.log(resp);
}
```

Attaching the log file also.
[sampleV.log](https://github.com/getnamo/socketio-client-ue4/files/1451671/sampleV.log)
UE Version: 4.17.1
Plugin version: "0.6.4"

**Comment by getnamo** [comment]
```Event Callback``` function should have an ```SIO Json Value``` signature parameter, not ```String```. See if that changes things. If not, could you also specify the socket.io server version?

**Comment by ziliwap** [comment]
You are amazing my friend! Yes it did work when the function input was **SIO Json Value**. My initial thought was if I am simply sending a string why not use a string type. But I stand corrected. Thanks a lot!

**Comment by getnamo** [comment]
Great! 

I want to see though if I can't get this error to spit out a warning instead of crashing and auto-convert to other input types. I'll investigate this possibility  before closing the issue.

**Comment by getnamo** [comment]
Added support for how you thought it worked as 0.6.8: https://github.com/getnamo/socketio-client-ue4/commit/591c1507aab15a52e2d42900bc75f9dec78345ab

Now if you know what type of response you expect it will bind correctly, if it's not exactly correct it will try to auto-convert. E.g. sioobject-> string auto casting etc. If the signature isn't supported you should now get a warning instead of a crash.

**Comment by ziliwap** [comment]
Kudos! This makes it a lot more user-friendly. Thanks again for an amazing plugin.

**Comment by getnamo** [comment]
Still waiting on a few fixes before posting a 4.18 binary.
Closing issue. If you find any bugs with the new version consider re-opening.

**Comment by sleoniy** [comment]
> `Event Callback` function should have an `SIO Json Value` signature parameter, not `String`. See if that changes things. If not, could you also specify the socket.io server version?

I am running on UE 4.20. Thank you for this fix!
